### PR TITLE
Makes shuttles rebuild pipenets on move

### PR DESCRIPTION
**Pull request by afterthought2**
From some limiting testing, appears to work and is not perceptibly adding to cost. Obviously is worse with very large shuttles.

### CODE REVIEW

**Review by comma — DISMISSED** [review]

File: code/modules/shuttles/shuttle.dm [hunk 1]
```diff
@@ -243,8 +240,37 @@
...
+// Remove all powernets and pipenets that were affected, and rebuild them.
+/datum/shuttle/proc/handle_pipes_and_power_on_move(var/list/new_turfs)
+	var/list/powernets = list()	
 	var/list/cables = list()
+	var/list/pipenets = list()
+	var/list/pipelines = list()
+	var/list/pipes = list()
+
+	for(var/turf/T in new_turfs)
+		for(var/obj/structure/cable/cable in T)
```
Comment (MistakeNot4892) [inline code/modules/shuttles/shuttle.dm:271]: Would this be faster as `for(var/thing in T)` and then typechecking?

File: code/modules/shuttles/shuttle.dm [hunk 2]
```diff
@@ -253,14 +279,17 @@
...
 			propagate_network(C,C.powernet)
-
-	if(mothershuttle)
-		var/datum/shuttle/mothership = SSshuttle.shuttles[mothershuttle]
-		if(mothership)
-			if(current_location.landmark_tag == motherdock)
-				mothership.shuttle_area |= shuttle_area
-			else
-				mothership.shuttle_area -= shuttle_area
+	for(var/datum/pipe_network/pipenet in pipenets)
```
Comment (MistakeNot4892) [inline code/modules/shuttles/shuttle.dm]: Can you rely on `powernet` and `return_network()` to return correct instances? If so you could `as anything` these loops.

**Comment by MistakeNot4892** [comment]
Other than optimization notes on the loops this looks fine, happy to approve for merge if you don't want to do the loop stuff.

**Comment by afterthought2** [comment]
I don't know enough about powernets to know if it's safe to do `as anything` there, but I did fix up the pipe ones. The power I left the way it was.

**Review by MistakeNot4892 — DISMISSED** [review]

**Review by comma — APPROVED** [review]

**Pull request merged**